lms/djangoapps/completion/api/v0/views.py

```diff
@@ -0,0 +1,70 @@
+""" API v0 views. """
+import logging
+
+from django.contrib.auth.models import User
+from django.core.exceptions import ValidationError
+
+from rest_framework.views import APIView
+from rest_framework.response import Response
+
+from opaque_keys.edx.keys import CourseKey, UsageKey
+
+from lms.djangoapps.completion.models import BlockCompletion
+
+log = logging.getLogger(__name__)
+
+
+class CompletionBatchView(APIView):
+    permission_classes = []
+
+    def _validate(self, batch_object):
+        """
+        Performs validation on the batch object to make sure it is in the proper format.
+
+        Parameters:
+            batch_object: The object provided by the caller.
+
+        Raises:
+
+            KeyError:
+                If a key is not found in the object.
+
+            django.core.exceptions.ValidationError:
+                If a float is passed that is not between 0.0 and 1.0.
+        """
+        if 'user' not in batch_object:
+            raise KeyError("Key 'user' not found.")
+        if 'course_key' not in batch_object:
+            raise KeyError("Key 'course_key' not found.")
+        if 'blocks' not in batch_object:
+            raise KeyError("Key 'blocks' not found.")
+        for value in batch_object['blocks'].values():
+            if 1.0 < value < 0.0:
+                raise ValidationError("All block values must be between 1.0 and 0.0")
```

Comment:
Be consistent about what error gets raised.  ValidationError or KeyError.  (ValidationError is probably better.)

Reply:
? I am a bit confused. Different errors for different issues.

Reply:
Is there a need to differentiate these errors, beyond the message that is returned to the user?  I would think it would be more straightforward to simply catch a ValidationError, and return a 400 response with the included message.  Thus a consistent interface to the messages will be more useful here.  (Also, a validation may be automatically turned into a 400, while a KeyError is bound to return a 500, which is not what we want, so it would have to be manually converted).